lib/tests/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+enable_testing()
+
+# package tests into a library so they can be referenced in all_tests
+add_library(ziti-tunnel-sdk-c-integ-test-lib OBJECT
+        dns_test.cpp
```

Comment:
does DNS test have to be integ?

Reply:
Maybe not, but It was convenient to have a `tunneler_context` when calling `ziti_dns_setup` in the test.

Reply:
I do think this is the easiest way without some refactoring and exposing `seed_dns` (or a new function that is aware of the tun IP and does all of the dns server setup minus adding the intercept context for the DNS server) to the test.